src/Service/CodeDeploy/tests/Unit/Input/CreateDeploymentInputTest.php

```diff
@@ -0,0 +1,86 @@
+use AsyncAws\CodeDeploy\ValueObject\EC2TagFilter;
+use AsyncAws\CodeDeploy\ValueObject\EC2TagSet;
+use AsyncAws\CodeDeploy\ValueObject\GitHubLocation;
+use AsyncAws\CodeDeploy\ValueObject\RawString;
+use AsyncAws\CodeDeploy\ValueObject\RevisionLocation;
+use AsyncAws\CodeDeploy\ValueObject\S3Location;
+use AsyncAws\CodeDeploy\ValueObject\TargetInstances;
+use AsyncAws\Core\Test\TestCase;
+
+class CreateDeploymentInputTest extends TestCase
+{
+    public function testRequest(): void
+    {
+        self::fail('Not implemented');
+
+        $input = new CreateDeploymentInput([
+            'applicationName' => 'change me',
+            'deploymentGroupName' => 'change me',
+            'revision' => new RevisionLocation([
+                'revisionType' => 'change me',
+                's3Location' => new S3Location([
+                    'bucket' => 'change me',
+                    'key' => 'change me',
+                    'bundleType' => 'change me',
+                    'version' => 'change me',
+                    'eTag' => 'change me',
+                ]),
+                'gitHubLocation' => new GitHubLocation([
+                    'repository' => 'change me',
+                    'commitId' => 'change me',
+                ]),
+                'string' => new RawString([
+                    'content' => 'change me',
+                    'sha256' => 'change me',
+                ]),
+                'appSpecContent' => new AppSpecContent([
+                    'content' => 'change me',
+                    'sha256' => 'change me',
+                ]),
+            ]),
+            'deploymentConfigName' => 'change me',
+            'description' => 'change me',
+            'ignoreApplicationStopFailures' => false,
+            'targetInstances' => new TargetInstances([
+                'tagFilters' => [new EC2TagFilter([
+                    'Key' => 'change me',
+                    'Value' => 'change me',
+                    'Type' => 'change me',
+                ])],
+                'autoScalingGroups' => ['change me'],
+                'ec2TagSet' => new EC2TagSet([
+                    'ec2TagSetList' => [[new EC2TagFilter([
+                        'Key' => 'change me',
+                        'Value' => 'change me',
+                        'Type' => 'change me',
+                    ])]],
+                ]),
+            ]),
+            'autoRollbackConfiguration' => new AutoRollbackConfiguration([
+                'enabled' => false,
+                'events' => ['change me'],
+            ]),
+            'updateOutdatedInstancesOnly' => false,
+            'fileExistsBehavior' => 'change me',
+        ]);
+
+        // see https://docs.aws.amazon.com/codedeploy/latest/APIReference/API_CreateDeployment.html
+        $expected = '
+            POST / HTTP/1.0
+            Content-Type: application/x-amz-json-1.1
+
+            {
+            "change": "it"
+        }
```

Comment:
I now need to fill in this part, is that right?

Reply:
yeah, exactly. Basically, make the CI green :)
And I'll be happy to merge it

Reply:
Any way to run tests locally?

Reply:
> 

Never mind, I was looking for PHPUnit in `vendor/bin`, but you guys have `simple-phpunit`.